src/datasource.test.ts

```diff
@@ -1584,60 +1584,123 @@ describe('OpenSearchDatasource', function(this: any) {
-        const query = ctx.ds.addAdHocFilters('', [
+        const query = ctx.ds.addAdHocFilters({ refId: 'A', query: '', queryType: QueryType.Lucene }, [
           { key: 'field:name', operator: '=', value: 'field:value', condition: '' },
         ]);
         expect(query).toBe('field\\:name:"field:value"');
       });
     });
 
     describe('with multiple ad hoc filters', () => {
-      const adHocFilters = [
-        { key: 'bar', operator: '=', value: 'baz', condition: '' },
-        { key: 'job', operator: '!=', value: 'grafana', condition: '' },
-        { key: 'service', operator: '=~', value: 'service', condition: '' },
-        { key: 'count', operator: '>', value: '1', condition: '' },
-      ];
-      it('should correctly add ad hoc filters when query is not empty', () => {
-        const query = ctx.ds.addAdHocFilters('foo:"bar" AND test:"test1"', adHocFilters);
-        expect(query).toBe(
-          'foo:"bar" AND test:"test1" AND bar:"baz" AND -job:"grafana" AND service:/service/ AND count:>1'
-        );
+      describe('Lucene queries', () => {
+        const adHocFilters = [
+          { key: 'bar', operator: '=', value: 'baz', condition: '' },
+          { key: 'job', operator: '!=', value: 'grafana', condition: '' },
+          { key: 'service', operator: '=~', value: 'service', condition: '' },
+          { key: 'count', operator: '>', value: '1', condition: '' },
+        ];
+
+        it('should correctly add ad hoc filters when query is not empty', () => {
+          const query = ctx.ds.addAdHocFilters(
+            { refId: 'A', query: 'foo:"bar" AND test:"test1"', queryType: QueryType.Lucene },
+            adHocFilters
+          );
+          expect(query).toBe(
+            'foo:"bar" AND test:"test1" AND bar:"baz" AND -job:"grafana" AND service:/service/ AND count:>1'
+          );
+        });
+
+        it('should correctly add ad hoc filters when query is  empty', () => {
+          const query = ctx.ds.addAdHocFilters({ refId: 'A', query: '', queryType: QueryType.Lucene }, adHocFilters);
+          expect(query).toBe('bar:"baz" AND -job:"grafana" AND service:/service/ AND count:>1');
+        });
       });
 
-      it('should correctly add ad hoc filters when query is  empty', () => {
-        const query = ctx.ds.addAdHocFilters('', adHocFilters);
-        expect(query).toBe('bar:"baz" AND -job:"grafana" AND service:/service/ AND count:>1');
+      describe('PPL queries', () => {
+        const adHocFilters = [
+          { key: 'bar', operator: '=', value: 'baz', condition: '' },
+          { key: 'job', operator: '!=', value: 'grafana', condition: '' },
+          { key: 'bytes', operator: '>', value: 50, condition: '' },
+          { key: 'count', operator: '<', value: 100, condition: '' },
+          { key: 'timestamp', operator: '=', value: '2020-11-22 16:40:43', condition: '' },
+        ];
+        it('should return query with ad-hoc filters applied', () => {
```

Comment:
Should this also have the query is empty case or is that not possible for ppl queries?

Reply:
right now the flow of the PPL execution is to always make sure the query string is not empty by adding the index that is being searched to the query string if the query string is empty

https://github.com/grafana/opensearch-datasource/blob/6e6f9affd5bfbf66ac939a90768782dc7b176ce3/src/datasource.ts#L741-L743

this happens happens before ad-hoc filters are added to the query string, so for this PR, i'm just testing the adding of the ad-hoc filters.

an empty query string test case will probably be more suited for https://github.com/grafana/opensearch-datasource/issues/200

in short, ppl queries should not be empty, but that should be handled outside of the `addAdHocFilters` code.